### ISSUE TITLE
Build protobuf from CMake

### ISSUE THREAD

**Issue by bcebere** [OpenMined/TenSEAL#182]
## Description
Now we expect to find protobuf already installed.
We should build libprotobuf and protoc from cmake, and use those.

**Comment by desilo-seungwoojung** [comment]
Hi, I am trying TenSEAL for the first time to use newly introduced CKKSTensor, but I think I am encountering the issue you are describing:

```
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libprotobuf.a(arena.o): relocation R_X86_64_TPOFF32 against symbol `_ZN6google8protobuf8internal9ArenaImpl13thread_cache_E' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libprotobuf.a(common.o): relocation R_X86_64_PC32 against symbol `stderr@@GLIBC_2.2.5' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/_tenseal_cpp.dir/build.make:177: _tenseal_cpp.cpython-38-x86_64-linux-gnu.so] Error 1
make[1]: *** [CMakeFiles/Makefile2:297: CMakeFiles/_tenseal_cpp.dir/all] Error 2
[100%] Built target _sealapi_cpp
make: *** [Makefile:130: all] Error 2
```

It seems that libprotobuf 3.6.1 installed via apt is not compiled with -fPIC flag, which is causing linking to fail. If you would like to, I can take a stab at making the change.

**Comment by bcebere** [comment]
Hi. That sounds great, thank you!

The CMAKE build should reproduce the steps from here for building protobuf https://github.com/OpenMined/TenSEAL/blob/master/.github/workflows/scripts/install_req_ubuntu.sh

Let us know if need anything!

**Comment by bcebere** [comment]
@desilo-seungwoojung have you made any progress with this issue?

we would love to include the fix in the 0.3.0 release.

**Comment by desilo-seungwoojung** [comment]
understood, will give you a pull request in a couple of days!
